main/main.cpp

```diff
@@ -276,6 +276,7 @@ void Main::print_help(const char *p_binary) {
 	OS::get_singleton()->print("\n");
 
 	OS::get_singleton()->print("Standalone tools:\n");
+	OS::get_singleton()->print("  -i, --init <name> [<path>]       Create a new project.\n");
```

Comment:
we already have a --path parameter. Should we reuse that instead?

Reply:
That would make it more semantic.
I wonder if we should have a CLI parser to validate input and format so we can separate that from the logic based on the content. The actual code handling the cli arguments feels a bit convoluted and hardcoded.
That would make easier to have proper error handling and not opening the editor when you pass an invalid argument.

Reply:
@lupoDharkael See #20000 :)

Reply:
Hmm, mixing `godot -i test test` and I don't really know which one is path and which one is project name, so seems logical...

To salvage the problem with whether it should create a folder by default, maybe this:
`godot --init test --path` *(would this be empty?)*

would allow to create project named `test` and making a directory for it, while
`godot --init test --path .`

would create a project inside current directory.
 
Or just omitting the path argument would force to create a folder named the same as project:
`godot --init test`

Reply:
I agree that the `--init` command should not take an optional path argument.
It should also not default to creating a folder, unless the path given does not exist, in which .ase it should try to create it.

So:
```
// Create "test" project in current working directory
godot --init test
// Create "test" project in previously created "myfolder" directory
mkdir myfolder
godot --init test --path myfolder
// Create "test" project in myother/folder subdirectory, creating them as needed
godot --init test --path myother/folder
```